### ISSUE TITLE
missing database binary file in .meta 

### ISSUE THREAD

**Issue by clb21565** [Nesvilab/philosopher#97]
Dear Philosopher team, 

Thank you for your very helpful tools! 

I am trying to process some comet searches I did before I knew of Philosopher, but I used the Comet decoy = 1 setting where decoy targets are made on the fly, so my database does not have the decoys listed within the fasta. 

I have managed to successfully (?) process my samples up until the filtering step, where I am getting the below error back - (note I am running this on a HPC cluster): 

time="12:07:12" level=info msg="Executing Workspace  v2.0.0"
time="12:07:12" level=warning msg="Can't check for updates, server unreachable"
time="12:07:12" level=info msg="Creating workspace"
time="12:07:12" level=warning msg="A meta data folder was found and will not be overwritten. "
time="12:07:12" level=info msg=Done
time="12:07:12" level=info msg="Executing Filter  v2.0.0"
time="12:07:12" level=info msg="Processing peptide identification files"
time="12:07:22" level=info msg="1+ Charge profile" decoy=0 target=0
time="12:07:22" level=info msg="2+ Charge profile" decoy=0 target=0
time="12:07:22" level=info msg="3+ Charge profile" decoy=0 target=1091
time="12:07:22" level=info msg="4+ Charge profile" decoy=0 target=1082
time="12:07:22" level=info msg="5+ Charge profile" decoy=0 target=200
time="12:07:22" level=info msg="6+ Charge profile" decoy=0 target=0
time="12:07:22" level=info msg="Database search results" ions=1379 peptides=1363 psms=2373
time="12:07:22" level=info msg="Converged to 0.00 % FDR with 2373 PSMs" decoy=0 threshold=0.0504 total=2373
time="12:07:23" level=info msg="Converged to 0.00 % FDR with 1363 Peptides" decoy=0 threshold=0.0504 total=1363
time="12:07:23" level=info msg="Converged to 0.00 % FDR with 1379 Ions" decoy=0 threshold=0.0504 total=1379
time="12:07:24" level=info msg="Protein inference results" decoy=0 target=890
time="12:07:24" level=info msg="Converged to 0.00 % FDR with 736 Proteins" decoy=0 threshold=0.0503 total=736
time="12:07:24" level=info msg="2D FDR estimation: Protein mirror image" decoy=736 target=736
time="12:07:25" level=info msg="Second filtering results" ions=860 peptides=847 psms=1761
time="12:07:25" level=info msg="Converged to 0.00 % FDR with 1761 PSMs" decoy=0 threshold=0.0504 total=1761
time="12:07:25" level=info msg="Converged to 0.00 % FDR with 847 Peptides" decoy=0 threshold=0.0504 total=847
time="12:07:25" level=info msg="Converged to 0.00 % FDR with 860 Ions" decoy=0 threshold=0.0504 total=860
time="12:07:25" level=warning msg="Cannot marshal file. open .meta/db.bin: no such file or directory"
time="12:07:25" level=warning msg="Cannot serialize file. EOF"
time="12:07:25" level=fatal msg="Database data not available, interrupting processing"

I am planning currently to repeat my searches through philosopher, but would you mind clarifying where this issue is arising? I am suspecting it is either the absence of DECOY tags in the FASTA DB or due to the fact that I ran comet independent of philosopher.

Also, is there a way to get around this without repeating searches? My database is quite large and the searches take quite a bit of computing time. 

Again, so thankful for these easy-access tools. Such a rarity in proteomics, please keep up the excellent, important work! 

Warmest Regards from Virginia,

Connor

**Comment by prvst** [comment]
Thanks for the kind words.

Can you share with me your database file and one pepXML file? If you don't want to post it here I can send you a Box share request.

**Comment by clb21565** [comment]
Hi there, the database is the UniRef100 (sorry, I understand that is probably beyond the scope of what these tools were intended to work with). You should be able to access it through the FTP at UniProt here https://www.uniprot.org/downloads

I have uploaded my .pep.xml file  along with this response 

[CP1.pep.xml.zip](https://github.com/Nesvilab/philosopher/files/4086814/CP1.pep.xml.zip)

**Comment by prvst** [comment]
Add the following tag to your fitler command: `--tag DECOY` and try again

**Comment by clb21565** [comment]
Resubmitted the job and the problem persists: 

time="10:23:05" level=info msg=Done
time="10:23:05" level=info msg="Executing Filter  v2.0.0"
time="10:23:05" level=info msg="Processing peptide identification files"
time="10:23:15" level=info msg="1+ Charge profile" decoy=0 target=0
time="10:23:15" level=info msg="2+ Charge profile" decoy=0 target=0
time="10:23:15" level=info msg="3+ Charge profile" decoy=224 target=867
time="10:23:15" level=info msg="4+ Charge profile" decoy=302 target=780
time="10:23:15" level=info msg="5+ Charge profile" decoy=82 target=118
time="10:23:15" level=info msg="6+ Charge profile" decoy=0 target=0
time="10:23:15" level=info msg="Database search results" ions=1379 peptides=1363 psms=2373
time="10:23:15" level=info msg="Converged to 0.00 % FDR with 0 PSMs" decoy=0 threshold=10 total=0
time="10:23:15" level=info msg="Converged to 0.00 % FDR with 0 Peptides" decoy=0 threshold=10 total=0
time="10:23:15" level=info msg="Converged to 0.00 % FDR with 0 Ions" decoy=0 threshold=10 total=0
time="10:23:16" level=info msg="Protein inference results" decoy=0 target=890
time="10:23:16" level=info msg="Converged to 0.00 % FDR with 736 Proteins" decoy=0 threshold=0.0503 total=736
time="10:23:16" level=info msg="2D FDR estimation: Protein mirror image" decoy=736 target=736
time="10:23:16" level=info msg="Second filtering results" ions=860 peptides=847 psms=1761
time="10:23:16" level=info msg="Converged to 0.00 % FDR with 1761 PSMs" decoy=0 threshold=0.0504 total=1761
time="10:23:17" level=info msg="Converged to 0.00 % FDR with 847 Peptides" decoy=0 threshold=0.0504 total=847
time="10:23:17" level=info msg="Converged to 0.00 % FDR with 860 Ions" decoy=0 threshold=0.0504 total=860
time="10:23:17" level=warning msg="Cannot marshal file. open .meta/db.bin: no such file or directory"
time="10:23:17" level=warning msg="Cannot serialize file. EOF"
time="10:23:17" level=fatal msg="Database data not available, interrupting processing"

**Comment by prvst** [comment]
sorry then, I'm afraid you will have to start over again using a preformatted database if you want to use our tools for downstream analysis. MSFRagger is extremely fast, that could help as well.

**Comment by clb21565** [comment]
Understood, I appreciate the feedback. One follow-up: I am having trouble getting my database through philosopher database due to the memory requirements for processing it (it's 106gb). Would you have a recommendation for how much memory to request to process such a database?

**Comment by prvst** [comment]
the basal memory footprint for philosopher is ~ 500MB. The real usage will depend on how much data you have to process. If your data / database are consuming too much memory from your computer, then you might need to use one with larger specs.

Are you working with metagenomics / metaproteomics ?

**Comment by clb21565** [comment]
I am running philosopher through the interface to an institutional HPC cluster, so I have access to a great deal of memory. I have tried running it requesting 2500 and 5000 MB across multiple nodes, but this failed with an OOM error. Presently, it is running on 32 nodes of the HPC with a request for the max memory per node.... no crash so far (45 minutes in). 

Yes, this is a metaproteomic search with a very large database (106GB).

**Comment by clb21565** [comment]
Dear Felipe, just touching base about this again. 

I have tried repeating the database step with a smaller database size (~15gb), yet this is still too big for 6 nodes with default memory settings on our HPC cluster, and I received an OOM error from the run. I took a look at the memory usage and it seems quite high: 10.84 GB
 
FYI, these are the settings I used: 

philosopher database  --custom /work/cascades/clb21565/home/Databases/uniref50.fasta --contam

I am going to repeat this run with max memory requested (across  6 nodes) but I wonder if there is a way to make this step more memory efficient. 

Again, very thankful for these tools. I believe that this package is probably the best option for metaproteomic data analysis out there at the moment. 

Best regards, 

Connor

**Comment by prvst** [comment]
Have you tried running this step outside the cluster ? This is only required to be executed once, if you have a way to run this is a different environment, you can then move the binary data to your cluster.

**Comment by clb21565** [comment]
Dear Felipe, I'm marking this as resolved because I have figured out a work around to problems encountered in prepping the database. A solution my team came up with follows below: 

First, yes, this could be run on a solid workstation but I elected not to do that as I would like to be able to pass this workflow to collaborators around campus where they can process their own database. 

So what I am planning to do is to create a docker container that can be run with Singularity on our cluster. This will combat a couple problems. 

Originally, I was running out of memory in trying to process my large databases - I finally was able to process one 15gb database successfully by using cores with more memory. However, when I tried to process another database (~46gb) the /tmp/ file that is shared by all users of our clusters was running out of space to write the database file. 

So, this is where the docker container comes in. Using this, the plan is to make a container where I can bind mount a /tmp/ file to an SSD on one of the cores. This way, the job will have more than enough hard drive space to write the tmp file. 

Appreciate the feedback from you, and just wanted to share my troubleshooting in case any one needs to follow in my footsteps!
